### PR TITLE
Changed Collection of Entities

### PR DESCRIPTION
**Pull request by Iuke121**
Revisit grabbing entities with TryHashSet.
This change includes minion entities parsing from https://github.com/shalzuth/LostArkLogger/pull/24

### CODE REVIEW

File: LostArkLogger/Parser.cs [hunk 1]
```diff
@@ -87,23 +86,20 @@ void ProcessPacket(List<Byte> data)
...
-                    IdToName[pc.PlayerId] = pc.Name + " (" + pcClass + ")";
+                    Entities.Add(new Entity { EntityId = pc.PlayerId, Name = pc.Name, ClassName = Npc.GetPcClass(pc.ClassId), Type = Entity.EntityType.Player });
                 }
                 else if (opcode == OpCodes.PKTInitEnv)
                 {
+                    Entities.Clear();
                     var pc = new PKTInitEnv(payload);
-                    IdToName[pc.PlayerId] = "You";
+                    Console.WriteLine("New instance, your id: " + pc.PlayerId);
+                    Entities.Add(new Entity { EntityId = pc.PlayerId, Name = "YOU" });
```
Comment (Edrolia) [inline LostArkLogger/Parser.cs:102]:
```suggestion
                    Entities.Add(new Entity { EntityId = pc.PlayerId, Name = "You" });
```

File: LostArkLogger/Parser.cs [hunk 2]
```diff
@@ -115,24 +111,32 @@ void ProcessPacket(List<Byte> data)
...
+                        { 
                             var skillName = Skill.GetSkillName(damage.SkillId, damage.SkillIdWithState);
-                            var ownerId = ProjectileOwner.ContainsKey(damage.PlayerId) ? ProjectileOwner[damage.PlayerId] : damage.PlayerId;
-                            var sourceName = IdToName.ContainsKey(ownerId) ? IdToName[ownerId] : ownerId.ToString("X");
-                            var destinationName = IdToName.ContainsKey(dmgEvent.TargetId) ? IdToName[dmgEvent.TargetId] : dmgEvent.TargetId.ToString("X");
-                            if (sourceName == "You" && Skill.GetClassFromSkill(damage.SkillId) != "UnknownClass")
+                            var className = Skill.GetClassFromSkill(damage.SkillId);
+                            if (skillName != "UnknownClass" && sourceEntity.ClassName == "")
+                                sourceEntity.ClassName = className;
+                            sourceName += (" (" + sourceEntity.ClassName + ")");
```
Comment (Edrolia) [inline LostArkLogger/Parser.cs:124]:
```suggestion
                            sourceName =  $"You ( { sourceEntity.ClassName } )";
```

Comment (Iuke121) [reply LostArkLogger/Parser.cs:124]: This renamed every player as "You"

**Comment by shalzuth** [comment]
I can't merge this due to conflicts.
I'm mobile so I can't verify, but I think the conflict is the minion packet that was added - since it is already in the main branch, you don't need to add it as well.

**Pull request merged**